docs/enhancements/backup-restore-support.md

```diff
@@ -0,0 +1,153 @@
+---
+title: backup-restore-support
+authors:
+  - "@danielerez"
+creation-date: 2024-08-08
+last-updated: 2024-08-08
+---
+
+# Backup/Restore Support
+
+## Summary
+
+The Assisted Installer should support backup/restore and disaster recovery scenarios, either using OADP (OpenShift API for Data Protection) for ACM (Advanced Cluster Management), or, using Kube-API.
+I.e. the assisted-service should be resilient in such scenarios which, for this context and effort, means that restored/moved spoke clusters should keep the same state and behave the same on the new hub cluster. 
+
+## Motivation
+
+Provide resiliency in the assisted-service for safe backup/restore flows, allowing spoke clusters to be used without any restriction after DR scenarios or moving between hubs.
+
+### Goals
+
+Spoke clusters (with/without BMHs) should have a consistent state and function properly, including the ability to add/remove nodes, after these scenarios: 
+* Spoke clusters restored on a passive hub with ACM/OADP functionality.
+* Spoke clusters moved between hubs using Kube-API.
+
+### Non-Goals
+
+* Improve import day2 cluster solution by getting full host data (e.g. using an agent on existing nodes).
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an Infrastructure Admin, I want to backup an installed spoke cluster from an active hub, including the associated BMHs (e.g. by using ACM/Kube-API/GitOps/etc).
+Then, I want to safely restore the cluster on a passive hub, including the BMHs that should keep the same state. I.e. the restored cluster should behave exactly as in the active hub with no restrictions (support day2 operations, etc).
+Note: Unbound hosts should still reboot using a new discovery ISO.
+
+#### Story 2
+
+As an Infrastructure Admin, I want to backup an installed spoke cluster from an active hub,
+without including the BMHs. 
+Then, I want to safely restore the cluster on a passive hub.
+Note: since no BMHs are included, unbound hosts should be manually booted using a new discovery ISO. 
+
+#### Story 3
+
+As an Infrastructure Admin, I want to restore a spoke cluster in non-installed status (draft/in-progress/etc). The cluster on target hub should, ideally, keep the same status. Or, at least, fail gracefully and let the user retry the installation.
```

Comment:
Question: If we back up and restore a spoke cluster where the installation hasn't started yet and we don't have the full inventory, will it impact our ability to start the installation? will we get a validation failure?

Reply:
Once the user restores all resources, the agents will be in "disconnected" state.  Once the hosts boot with a new ISO, the agents will populate the inventory, and installation can proceed.